napari/utils/colormaps/colormaps.py

```diff
@@ -45,6 +46,38 @@ def _all_rgb():
 LABRNG = LABMAX - LABMIN
 
 
+def convert_vispy_colormap(colormap, name=None):
```

Comment:
just personal preference... but since this is basically just a modified `Colormap` constructor, I think it would be lovely for it to live on the `Colormap` (or at least be a private function if you really want it outside the class):

```python
class Colormap:
    @classmethod
    def from_vispy(cls, colormap, name=None):
        ...
        return cls(
            name=name,
            colors=colormap.colors.rgba,
            controls=controls,
            interpolation=interpolation,
        )
```

that cleans up imports elsewhere, so it's always:
```python
from napari.utils import Colormap

cmap = Colormap.from_vispy(vispy_colormap)
```
rather than sometimes importing Colormap and other times importing `convert_vispy_colormap` depending on the starting material.

also, having `name=None` here is going to break the typing on our Colormap object.  for example, if a name is not provided, and someone tries to do something like `cmap.name.lower()`, it would now be an attribute error. Either add `name=vispy_colormap` or something here... or make the default of `Colormap` be `None` and then set the default to `custom_colormap` in the `Colormap.__init__` function

Reply:
I think making a `Colormap.from_vispy()` constructor is a good idea!